File: android/app/src/main/java/com/starstorm/beer/service/ParseFriendService.java
package com.starstorm.beer.service;

import com.parse.FunctionCallback;
import com.parse.ParseCloud;
import com.parse.ParseObject;
import com.starstorm.beer.util.ParseHelper;

import java.util.HashMap;

/**
 * Created by Conor on 16/10/2014.
 */
public enum ParseFriendService implements FriendService {
    INSTANCE;

    @Override
    public void sendUnfriendRequest(ParseObject friendship, FunctionCallback<Object> callback) {
        HashMap<String, Object> params = ParseHelper.getDefaultParams();
        params.put("friendshipId", friendship.getObjectId());
        ParseCloud.callFunctionInBackground("unfriend", params, callback);
    }

    @Override
    public void sendFriendRequest(String username, FunctionCallback<Object> callback) {
        HashMap<String, Object> params = ParseHelper.getDefaultParams();
        params.put("username", username.toLowerCase());
        ParseCloud.callFunctionInBackground("sendfriendrequest", params, callback);
    }
}

File: android/app/src/main/kotlin/com/starstorm/beer/service/ParseFriendService.kt
package com.starstorm.beer.service

import com.parse.FunctionCallback
import com.parse.ParseCloud
import com.parse.ParseObject
import com.starstorm.beer.util.getDefaultParams

/**
 * Created by Conor on 16/10/2014.
 */
public enum class ParseFriendService {
    INSTANCE

    public fun sendUnfriendRequest(friendship: ParseObject, callback: FunctionCallback<Any>) {
        val params = getDefaultParams()
        params.put("friendshipId", friendship.getObjectId())
        ParseCloud.callFunctionInBackground<Any>("unfriend", params, callback)
    }

    public fun sendFriendRequest(username: String, callback: FunctionCallback<Any>) {
        val params = getDefaultParams()
        params.put("username", username.toLowerCase())
        ParseCloud.callFunctionInBackground<Any>("sendfriendrequest", params, callback)
    }
}
